src/db_adapters/receipts.rs

```diff
@@ -240,56 +344,68 @@ async fn find_tx_hashes_for_receipts(
             }
 
             receipts.retain(|r| {
-                !tx_hashes_for_receipts.contains_key(r.receipt_id.to_string().as_str())
+                !tx_hashes_for_receipts
+                    .contains_key(&crate::ReceiptOrDataId::ReceiptId(r.receipt_id))
             });
         }
 
-        let tx_hashes_for_receipts_via_outcomes: Vec<(
-            crate::ReceiptIdString,
-            crate::ParentTransactionHashString,
-        )> = crate::await_retry_or_panic!(
-            schema::execution_outcome_receipts::table
-                .inner_join(
-                    schema::receipts::table
-                        .on(schema::execution_outcome_receipts::dsl::executed_receipt_id
-                            .eq(schema::receipts::dsl::receipt_id)),
-                )
-                .filter(
-                    schema::execution_outcome_receipts::dsl::produced_receipt_id.eq(any(receipts
-                        .clone()
-                        .iter()
-                        .filter(|r| {
-                            matches!(
+        let tx_hashes_for_receipts_via_outcomes: Vec<(String, crate::ParentTransactionHashString)> =
+            crate::await_retry_or_panic!(
+                schema::execution_outcome_receipts::table
+                    .inner_join(
+                        schema::receipts::table
+                            .on(schema::execution_outcome_receipts::dsl::executed_receipt_id
+                                .eq(schema::receipts::dsl::receipt_id)),
+                    )
+                    .filter(
+                        schema::execution_outcome_receipts::dsl::produced_receipt_id.eq(any(
+                            receipts
+                                .clone()
+                                .iter()
+                                .filter(|r| {
+                                    matches!(
                                 r.receipt,
                                 near_indexer::near_primitives::views::ReceiptEnumView::Action { .. }
                             )
-                        })
-                        .map(|r| r.receipt_id.to_string())
-                        .collect::<Vec<crate::ReceiptIdString>>())),
-                )
-                .select((
-                    schema::execution_outcome_receipts::dsl::produced_receipt_id,
-                    schema::receipts::dsl::originated_from_transaction_hash,
-                ))
-                .load_async::<(crate::ReceiptIdString, crate::ParentTransactionHashString)>(pool),
-            10,
-            "Parent Transaction for Receipts were fetched".to_string(),
-            &receipts
-        )
-        .unwrap_or_default();
+                                })
+                                .map(|r| r.receipt_id.to_string())
+                                .collect::<Vec<String>>()
```

Comment:
Given that we introduced `ReceiptDataIdString` in this file, should we also re-introduce `ReceiptIdString` next to it or drop `ReceiptDataIdString`? Though, I see `ReceiptIdString` is also used in the `transactions.rs`, so we might as well just keep `ReceiptIdString` in main.rs

Reply:
No, I've dropped `ReceiptIdString` from `transactions.rs` and from the main. I'd vote for removing `ReceiptDataIdString` as well